homeassistant/components/overkiz/water_heater_entities/domestic_hot_water_production.py

```diff
@@ -0,0 +1,281 @@
+            self.executor.select_state(
+                OverkizState.CORE_MAXIMAL_TEMPERATURE_MANUAL_MODE
+            ),
+        )
+
+    @property
+    def target_temperature_low(self) -> float:
+        """Return the lowbound target temperature we try to reach."""
+        return cast(
+            float,
+            self.executor.select_state(
+                OverkizState.CORE_MINIMAL_TEMPERATURE_MANUAL_MODE
+            ),
+        )
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        target_temperature = kwargs.get(ATTR_TEMPERATURE)
+        await self.executor.async_execute_command(
+            OverkizCommand.SET_TARGET_TEMPERATURE, target_temperature
+        )
+
+        if self.executor.has_command(OverkizCommand.REFRESH_TARGET_TEMPERATURE):
+            await self.executor.async_execute_command(
+                OverkizCommand.REFRESH_TARGET_TEMPERATURE
+            )
+
+    @property
+    def current_operation(self) -> str:
+        """Return current operation ie. eco, electric, performance, ..."""
+        if self._is_boost_mode_on:
+            return OVERKIZ_TO_OPERATION_MODE[OverkizCommandParam.BOOST]
+
+        return OVERKIZ_TO_OPERATION_MODE[
+            cast(
+                str,
+                self.executor.select_state(
+                    OverkizState.IO_DHW_MODE, OverkizState.MODBUSLINK_DHW_MODE
+                ),
+            )
+        ]
+
+    async def async_set_operation_mode(self, operation_mode: str) -> None:
+        """Set new target operation mode."""
+
+        if operation_mode == STATE_PERFORMANCE:
+            if self.executor.has_command(OverkizCommand.SET_BOOST_MODE):
+                await self.executor.async_execute_command(
+                    OverkizCommand.SET_BOOST_MODE, OverkizCommand.ON
+                )
+
+            if self.executor.has_command(OverkizCommand.SET_BOOST_MODE_DURATION):
+                await self.executor.async_execute_command(
+                    OverkizCommand.SET_BOOST_MODE_DURATION, 7
+                )
+                await self.executor.async_execute_command(
+                    OverkizCommand.REFRESH_BOOST_MODE_DURATION
+                )
+
+            if self.executor.has_command(OverkizCommand.SET_CURRENT_OPERATING_MODE):
+                current_operating_mode = self.executor.select_state(
+                    OverkizState.CORE_OPERATING_MODE
+                )
+
+                if current_operating_mode and isinstance(current_operating_mode, dict):
+                    await self.executor.async_execute_command(
+                        OverkizCommand.SET_CURRENT_OPERATING_MODE,
+                        {
+                            OverkizCommandParam.RELAUNCH: OverkizCommandParam.ON,
+                            OverkizCommandParam.ABSENCE: OverkizCommandParam.OFF,
+                        },
+                    )
+
+            return
+
+        if self._is_boost_mode_on:
```

Comment:
Can you explain me why ?

I don't get this condition.

Why not using `operation_mode` ? or `else` ?

Reply:
at this point, we're setting an operation mode which is NOT boost (see previous if with a return)
This `if` check if the device is currently in BOOST mode
And all code inside this `if` disable boost in all possible way reardgind device.
I'm adding a simpler comment in the code to explain that.